### PR TITLE
Create Tests for 4.1 RC2 Code

### LINKED ISSUE #27554: Faces 4.1: Faces 4.1: Deprecate full state saving (FSS)
`https://github.com/jakartaee/faces/issues/1829`

### LINKED ISSUE #27562: Faces 4.1: id attribute is missing in vdl of h:head and h:body
`https://github.com/jakartaee/faces/issues/1760`

### LINKED ISSUE #27572: Faces 4.1: Add rowStatePreserved property to UIRepeat, exactly the same as UIData
`https://github.com/jakartaee/faces/issues/1263`

### PR DESCRIPTION
**Pull request by volosied**
Tests part of #28328

fixes https://github.com/OpenLiberty/open-liberty/issues/27554
fixes https://github.com/OpenLiberty/open-liberty/issues/27572
fixes https://github.com/OpenLiberty/open-liberty/issues/27562

### CODE REVIEW

**Comment by volosied** [comment]
#build

File: dev/io.openliberty.org.apache.myfaces.4.1_fat/fat/src/io/openliberty/org/apache/myfaces41/fat/test/RenderIdTest.java [hunk 1]
```diff
@@ -79,22 +88,124 @@ public void setupPerTest() throws Exception {
...
         try (WebClient webClient = new WebClient()) {
-            URL url = JSFUtils.createHttpUrl(server, APP_NAME, "index.xhtml");
+            URL url = JSFUtils.createHttpUrl(server, APP_NAME, "HTML5_withIDs.xhtml");
 
             HtmlPage page = (HtmlPage) webClient.getPage(url);
 
-            // assertNotNull("Id attribute not found on head!", page.getHead().getId());
+            logPage(page);
+
+            assertEquals("Id attribute was found on head!", "headID", page.getHead().getId());
```
Comment (pnicolucci) [inline dev/io.openliberty.org.apache.myfaces.4.1_fat/fat/src/io/openliberty/org/apache/myfaces41/fat/test/RenderIdTest.java]: Shouldn't the assert message be something like "Id attribute should have been found on head but was not!"?

File: dev/io.openliberty.org.apache.myfaces.4.1_fat/fat/src/io/openliberty/org/apache/myfaces41/fat/test/RenderIdTest.java [hunk 2]
```diff
@@ -79,22 +88,124 @@ public void setupPerTest() throws Exception {
...
+    @Test
+    public void verifyIDsAreGenerated_WithIDs_HTML4() throws Exception {
+        try (WebClient webClient = new WebClient()) {
+            URL url = JSFUtils.createHttpUrl(server, APP_NAME, "HTML4_withIDs.xhtml_xhtml");
+
+            HtmlPage page = (HtmlPage) webClient.getPage(url);
+
+            logPage(page);
+
+            assertEquals("Id attribute not found on head!", "", page.getHead().getId());
```
Comment (pnicolucci) [inline dev/io.openliberty.org.apache.myfaces.4.1_fat/fat/src/io/openliberty/org/apache/myfaces41/fat/test/RenderIdTest.java]: Shouldn't the message be "Id attribute was found on head and should not have been!"?

File: dev/io.openliberty.org.apache.myfaces.4.1_fat/fat/src/io/openliberty/org/apache/myfaces41/fat/test/RenderIdTest.java (hunk 1 above)
Comment (pnicolucci) [reply dev/io.openliberty.org.apache.myfaces.4.1_fat/fat/src/io/openliberty/org/apache/myfaces41/fat/test/RenderIdTest.java]: Please review your messages in this class as I think they need some work I won't comment on everyone.

File: dev/io.openliberty.org.apache.myfaces.4.1_fat/publish/servers/faces41_rowStatePreserved/server.xml [hunk 3]
```diff
@@ -0,0 +1,20 @@
+<!--
+    Copyright (c) 2024 IBM Corporation and others.
+    All rights reserved. This program and the accompanying materials
+    are made available under the terms of the Eclipse Public License 2.0
+    which accompanies this distribution, and is available at
+    http://www.eclipse.org/legal/epl-2.0/
+    
+    SPDX-License-Identifier: EPL-2.0
+ -->
+<server description="Server for testing Faces 4.1 ID Attribute Rendering">
```
Comment (pnicolucci) [inline dev/io.openliberty.org.apache.myfaces.4.1_fat/publish/servers/faces41_rowStatePreserved/server.xml]: I think this description needs to be updated for this server.

**Review by pnicolucci — APPROVED** [review]

**Comment by volosied** [comment]
#libby

**Pull request merged**